test/Tests/src/Network/Http_Spec.enso

```diff
@@ -1,332 +1,205 @@
+            HTTP.fetch url . to_file f
+            f.size . should_equal 7168
+            f.delete
+            f.exists.should_be_false
+
+        Test.specify "Can skip auto-parse" <|
+            response = HTTP.fetch url_get try_auto_parse_response=False
+            response.bytes.length . should_equal 272
+
+        Test.specify "unknown host" <|
+            HTTP.fetch "http://undefined_host.invalid" . should_fail_with Request_Error
+
+        Test.specify "Fails on a bad URL scheme" <|
+            HTTP.fetch "zxcv://bad.scheme" . should_fail_with Request_Error
+            HTTP.fetch "" . should_fail_with Request_Error
+
+    Test.group "post" pending=pending_has_url <|
+        base_url_with_slash = if base_url.ends_with "/" then base_url else base_url + "/"
+        url_post = base_url_with_slash + "post"
+
+        Test.specify "Can perform a Request_Body.Text POST" <|
+            response = HTTP.post url_post (Request_Body.Text "hello world")
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+
+        Test.specify "Can perform a Request_Body.Json JSON POST" <|
+            json = Json.parse '{"a": "asdf", "b": 123}'
+            response = HTTP.post url_post (Request_Body.Json json)
+            response.at "data" . should_equal '{"a":"asdf","b":123}'
+            response.at "headers" . at "Content-Length" . should_equal "20"
+            response.at "headers" . at "Content-Type" . should_equal "application/json"
+
+        Test.specify "Can perform a JSON POST" <|
+            json = Json.parse '{"a": "asdf", "b": 123}'
+            response = HTTP.post url_post json
+            response.at "data" . should_equal '{"a":"asdf","b":123}'
+            response.at "headers" . at "Content-Length" . should_equal "20"
+            response.at "headers" . at "Content-Type" . should_equal "application/json"
+
+        Test.specify "Can perform an object Request_Body.Json POST" <|
+            response = HTTP.post url_post (Request_Body.Json (Test_Type.Aaa "abc"))
+            response.at "data" . should_equal '{"type":"Test_Type","constructor":"Aaa","s":"abc"}'
+            response.at "headers" . at "Content-Length" . should_equal "50"
+            response.at "headers" . at "Content-Type" . should_equal "application/json"
+
+        Test.specify "Can perform am object JSON POST" <|
+            response = HTTP.post url_post (Test_Type.Bbb 12)
+            response.at "data" . should_equal '{"type":"Test_Type","constructor":"Bbb","i":12}'
+            response.at "headers" . at "Content-Length" . should_equal "47"
+            response.at "headers" . at "Content-Type" . should_equal "application/json"
+
+        Test.specify "can handle a bad .to_json" <|
+            HTTP.post url_post (Bad_To_Json.Aaa "abcd") . should_fail_with Illegal_Argument
```

Comment:
I'm wondering if this should be a `Type_Error`? Seems more fitting.

Reply:
This is to catch a buggy implementation of `to_json`, which is more like an `Illegal_Argument`, but perhaps there is a better one? I don't think it's a `Type_Error`.

Reply:
The bug comes from `to_json` returning a wrong type - we are expecting it to return `Text` and it returns _something else_. That smells like `Type_Error` to me.

Reply:
The bug in this test comes from `from_pairs` receiving data of the wrong type, and throwing `Illegal_Argument`. It is meant to simulate a generic error inside `to_json`, not a specific one like `Type_Error`. I'll change the test class to just throw the error.